src/node.rs

```diff
@@ -976,6 +973,23 @@ impl PbftNode {
             )));
         }
 
+        // Verify the seal itself
+        self.verify_consensus_seal(&seal, block.signer_id.clone(), state)?;
+
+        Ok(())
+    }
+
+    /// Verify the given consenus seal
+    ///
+    /// # Panics
+    /// + If the node is unable to query the validator for on-chain settings
```

Comment:
Won't this crash the consensus engine? I understand this is already the current behavior, but it likely means that the consensus engine will crash if the validator crashes as well.

Reply:
Yes, this is true. I think the correct way to handle this would be to try again until the validator responds. However, I think that this may actually be getting the peers from the wrong block; it should be getting the peers from the block previous to the one the seal is for, since that's when the votes were actually cast. I will need to rethink that bit, but the question about how to properly handle when the validator isn't available still stands.